tests/parametric/test_headers_precedence.py

```diff
@@ -93,6 +93,7 @@ class Test_Headers_Precedence:
     @missing_feature(context.library == "php", reason="New 'datadog' default hasn't been implemented yet")
     @missing_feature(context.library == "python", reason="New 'datadog' default hasn't been implemented yet")
     @missing_feature(context.library == "python_http", reason="New 'datadog' default hasn't been implemented yet")
+    @bug(context.library == "java", reason="Does not inject traceparent")
```

Comment:
This might require multiple lines to satisfy the linter, but I think we can do a version check on this instead. And I think we'll want to do this version check for Ruby as well @marcotc 

```suggestion
    @irrelevant(context.library < "java@1.24.0", reason="Newer versions include tracecontext as a default propagator")
```

Reply:
If we need to unblock a test failure on main we can skip this suggestion and apply it in a followup PR